text/0001-dynamic-fees-v1.md

```diff
@@ -0,0 +1,93 @@
+
+Because of this, two nodes with different fees will produce different encryption keys, which will lead to a failed decyption during the key exchange setup (specifically, the initiator will receive an `AeadError` when attempting to decrypt the initial message received from the responder).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The primary drawback to this approach is that it requires a restart of a quorum of the network in order to adjust to a new fee. In the intended common-case (low fees generally, higher fees in response to an active denial-of-service), this will entail restarting the network twice, and when a node is restarted, any pending transactions which did not make it into the current SCP slot will be discarded.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The first alternative considered was simply lowering the fees to 400uMOB, or 0.02USD @ 50USD/MOB. As noted above, while this accomplishes the initial goal of lowering the fees, it does not provide the ability to raise them without another enclave upgrade, which is costly, time-consuming, and will take upwards of 120 days to complete.
+
+The next alternative was using a "fee schedule", or "fee calendar"---that is, rather than having a single fee which is constructed at startup time, use a configuration file which provides a series of "fee ranges" `(fee, startBlock, endBlock)` that are hashed into the key exchange.
+
+This alternative scheme does have the advantage of allowing a reconfiguration to "script" the denial-of-service response, and it's subsequent unwind, in a single restart. The devil, however, is in the details: how should pending and future transactions be handled?
+
+This is actually quite tricky to get right: obviously you want to immediately reject any transactions which do not meet the currently configured fee requirements, but it's entirely possible that your node is already queuing transactions when this comes in.
+
+In this scenario, a client submits a transaction, and it goes into a queue behind hundreds of other transactions. The fee is "correct" for the current block, but the transaction will not actually be entertained for acceptance until after the fee has been raised, and the fee is no longer valid.
+
+Chosing to simply accept that if any node has accepted a proposed transaction at a lower fee, that fee is (or would have been) valid for all nodes is an option, but this does make the ability to enforce fees dependent on the union of all SGX functionality, which is a change from our ad-hoc norms around the use of SGX as a privacy-enhancement technology, rather than a security-critical technology.
```

Comment:
```suggestion
Choosing to simply accept that if any node has accepted a proposed transaction at a lower fee, that fee is (or would have been) valid for all nodes is an option, but this does make the ability to enforce fees dependent on the union of all SGX functionality, which is a change from our ad-hoc norms around the use of SGX as a privacy-enhancement technology, rather than a security-critical technology.
```